internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -242,7 +242,7 @@ func (r *Reconciler) reconcileControlPlane(ctx context.Context, s *scope.Scope)
 
 		// Reconcile the current and desired state of the MachineHealthCheck.
 		if err := r.reconcileMachineHealthCheck(ctx, s.Current.ControlPlane.MachineHealthCheck, s.Desired.ControlPlane.MachineHealthCheck); err != nil {
-			return errors.Wrapf(err, "failed to update %s", tlog.KObj{Obj: s.Desired.ControlPlane.MachineHealthCheck})
+			return err
```

Comment:
I would also keep the error here, same as in https://github.com/kubernetes-sigs/cluster-api/pull/5971/files#r791625900 (below)

Although I would prefer to use "failed to reconcile" as we don't know if it's a create, update or delete afaik (would be nice if you can adjust other cases of `failed to update` when we're using it after a reconcile, too)